### PR TITLE
[cmake] Fix target onnxruntime_providers_cuda installation path

### PR DESCRIPTION
**Pull request by JackBoosY**
**Description**: Describe your changes.
The target `onnxruntime_providers_cuda` should always be built as dynamic and be used with dynamic load.
So the installation path should be `CMAKE_INSTALL_BINDIR` instead of `CMAKE_INSTALL_LIBDIR`.

Fix that.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @snnn

File: cmake/onnxruntime_providers.cmake [hunk 1]
```diff
@@ -502,7 +502,7 @@ if (onnxruntime_USE_CUDA)
 
   install(TARGETS onnxruntime_providers_cuda
           ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-          LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+          LIBRARY  DESTINATION ${CMAKE_INSTALL_BINDIR}
```
Comment (snnn) [inline cmake/onnxruntime_providers.cmake:505]: Does it mean on Linux it will be installed to /usr/local/bin instead of /usr/local/lib ?

Comment (JackBoosY) [reply cmake/onnxruntime_providers.cmake:505]: Yeah, I can add some code, wait for a while.

**Comment by WilliamTambellini** [comment]
RFC

**Comment by JackBoosY** [comment]
Ah this PR is too old, I can close this now.

**Pull request closed without merging**